### PR TITLE
Fix config

### PR DESCRIPTION
**Pull request by frankharkins**
### Summary

Fixes the config so the tests work on my machine.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,6 +4,7 @@ envlist = py39, lint, coverage, black
 skipsdist = True
 
 [testenv]
+basepython = python3.9
```
Comment (frankharkins) [inline tox.ini:7]: This fixed the following linting error:

```
lint run-test: commands[0] | pylint -rn ecosystem tests
Traceback (most recent call last):
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
             ^^^^^^^^^^^^
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/__init__.py", line 21, in run_pylint
    from pylint.lint import Run as PylintRun
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/lint/__init__.py", line 76, in <module>
    from pylint.lint.parallel import check_parallel
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/lint/parallel.py", line 7, in <module>
    from pylint import reporters
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/reporters/__init__.py", line 26, in <module>
    from pylint import utils
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/utils/__init__.py", line 46, in <module>
    from pylint.utils.ast_walker import ASTWalker
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/pylint/utils/ast_walker.py", line 6, in <module>
    from astroid import nodes
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/astroid/__init__.py", line 53, in <module>
    from astroid import inference, raw_building
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/astroid/inference.py", line 33, in <module>
    import wrapt
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/wrapt/__init__.py", line 10, in <module>
    from .decorators import (adapter_factory, AdapterFactory, decorator,
  File "/Users/frankharkins/Documents/ecosystem/.tox/lint/lib/python3.11/site-packages/wrapt/decorators.py", line 34, in <module>
    from inspect import ismethod, isclass, formatargspec
ImportError: cannot import name 'formatargspec' from 'inspect' (/usr/local/Cellar/python@3.11/3.11.7_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/inspect.py)
```

File: setup.cfg [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [metadata]
-description-file = README.md
+description_file = README.md
```
Comment (frankharkins) [inline setup.cfg:2]: This fixed the following error.

```
/Users/frankharkins/Documents/ecosystem/.tox/py39/lib/python3.9/site-packages/setuptools/dist.py:476: SetuptoolsDeprecationWarning: Invalid dash-separated options
!!

        ********************************************************************************
        Usage of dash-separated 'description-file' will not be supported in future
        versions. Please use the underscore name 'description_file' instead.

        By 2024-Sep-26, you need to update your project and remove deprecated calls
        or your builds will no longer be supported.

        See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
        ********************************************************************************

!!
  opt = self.warn_dash_deprecation(opt, section)
```

File: setup.py [hunk 3]
```diff
@@ -18,5 +18,5 @@
     packages=setuptools.find_packages(),
     package_data={"ecosystem": ["html_templates/*.jinja"]},
     install_requires=install_requires,
-    python_requires='>=3.6'
+    python_requires='>=3.9'
```
Comment (frankharkins) [inline setup.py:21]: Dropping support for Python 3.6 since we're only testing with 3.9 now.

File: tox.ini (hunk 1 above)
Comment (Eric-Arellano) [reply tox.ini:7]: Another approach could be updating pylint, and if they're pinned, astroid and wrapt.

**Review by Eric-Arellano — APPROVED** [review]
Looks like CI failure

File: tox.ini (hunk 1 above)
Comment (frankharkins) [reply tox.ini]: That's a better idea

**Comment by frankharkins** [comment]
I decided not to upgrade pylint as we'd need to update the codebase to fit its new rules. It's not super difficult but I want to merge this to unblock other PRs.

**Comment by Eric-Arellano** [comment]
> I decided not to upgrade pylint as we'd need to update the codebase to fit its new rules.

For future reference, it's acceptable imo to disable those new lints with a TODO comment saying they should be un-ignored.

**Pull request merged**